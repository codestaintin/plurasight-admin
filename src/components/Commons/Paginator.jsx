import React from 'react';
import Paginate from 'react-paginate';
import { number, func } from 'prop-types';

const Paginator = ({ pageCount, page, handlePageClick }) => (
  <Paginate
    pageCount={pageCount}
    pageRangeDisplayed={2}
    initialPage={page - 1}
    marginPagesDisplayed={2}
    disableInitialCallback
    onPageChange={handlePageClick}
    containerClassName="pagination justify-content-center"
    subContainerClassName="page-item"
    pageLinkClassName="page-link"
    activeClassName="page-item active"
    nextLinkClassName="page-link"
    previousLinkClassName="page-link"
  />
);

Paginator.propTypes = {
  page: number,
  pageCount: number,
  handlePageClick: func.isRequired
};
export default Paginator;
