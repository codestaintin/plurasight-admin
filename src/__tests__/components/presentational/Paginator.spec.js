import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import Paginator from '../../../components/Commons/Paginator';

configure({ adapter: new Adapter() });

describe('<Paginator>', () => {
  const props = {
    pageCount: 2,
    page: 1,
    handlePageClick: jest.fn(() => {})
  };

  it('renders the <Paginator> component', () => {
    const tree = shallow(<Paginator {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
