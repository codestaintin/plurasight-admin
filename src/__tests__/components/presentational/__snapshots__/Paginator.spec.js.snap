// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paginator> renders the <Paginator> component 1`] = `
<PaginationBoxView
  activeClassName="page-item active"
  breakLabel="..."
  containerClassName="pagination justify-content-center"
  disableInitialCallback={true}
  disabledClassName="disabled"
  initialPage={0}
  marginPagesDisplayed={2}
  nextClassName="next"
  nextLabel="Next"
  nextLinkClassName="page-link"
  onPageChange={[MockFunction]}
  pageCount={2}
  pageLinkClassName="page-link"
  pageRangeDisplayed={2}
  previousClassName="previous"
  previousLabel="Previous"
  previousLinkClassName="page-link"
  subContainerClassName="page-item"
/>
`;
